### PR TITLE
3/fix area menu close after add

### PR DESCRIPTION
**Pull request by stuartromanek**
https://apostrophecms.atlassian.net/jira/software/projects/A30U/boards/4?selectedIssue=A30U-913
- cleans up some unused event listeners
- manually closes context menu when a user adds something to a page/doc

As noted, context menus should really be able to handle this on their own but are currently relying on blurring or using the full `<AposContextMenu/>` implementation, which is being overridden and cherry picked here

### CODE REVIEW

File: modules/@apostrophecms/area/ui/apos/components/AposAreaMenu.vue [hunk 1]
```diff
@@ -4,7 +4,7 @@
       :button="buttonOptions"
       v-bind="extendedContextMenuOptions"
       @open="menuOpen"
-      @close="menuClose"
```
Comment (abea) [inline modules/@apostrophecms/area/ui/apos/components/AposAreaMenu.vue:7]: We're not using the close event here at all?

File: modules/@apostrophecms/area/ui/apos/components/AposAreaMenu.vue [hunk 2]
```diff
@@ -102,7 +102,7 @@ export default {
       type: Boolean
     }
   },
-  emits: [ 'menu-close', 'menu-open', 'insert' ],
```
Comment (abea) [inline modules/@apostrophecms/area/ui/apos/components/AposAreaMenu.vue:105]: `AposAreaWidget` is still listening for `@menu-close`

**Review by abea — CHANGES_REQUESTED** [review]

File: modules/@apostrophecms/area/ui/apos/components/AposAreaMenu.vue (hunk 1 above)
Comment (stuartromanek) [reply modules/@apostrophecms/area/ui/apos/components/AposAreaMenu.vue:7]: you are correct this is still in play, re implemented

File: modules/@apostrophecms/area/ui/apos/components/AposAreaMenu.vue (hunk 2 above)
Comment (stuartromanek) [reply modules/@apostrophecms/area/ui/apos/components/AposAreaMenu.vue:105]: re implemented

File: modules/@apostrophecms/area/ui/apos/components/AposAreaMenuItem.vue [hunk 3]
```diff
@@ -48,42 +48,8 @@ export default {
 </script>
 
 <style lang="scss" scoped>
-.apos-tag-list__button {
-  @include apos-button-reset();
-  display: flex;
-  align-items: center;
-  padding: 7.5px 10px;
-  border-radius: 5px;
```
Comment (stuartromanek) [inline modules/@apostrophecms/area/ui/apos/components/AposAreaMenuItem.vue:56]: unrelated cleanup of copypasta'd template guts

File: modules/@apostrophecms/area/ui/apos/components/AposAreaMenu.vue [hunk 4]
```diff
@@ -164,6 +165,10 @@ export default {
       this.$emit('menu-open', e);
     },
     async add(name) {
+      // Potential TODO: If we find ourselves manually flipping these bits in other AposContextMenu overrides
+      // we should consider refactoring contextmenus to be able to self close when any click takes place within their el
+      // as it is often the logical experience (not always, see tag menus and filters)
+      this.$refs.contextMenu.isOpen = false;
```
Comment (stuartromanek) [inline modules/@apostrophecms/area/ui/apos/components/AposAreaMenu.vue:171]: @abea I also changed this to be explicitly false (closed) because at this stage we always want it closed, truly toggling it felt overly cute and possibly bug prone

**Review by abea — APPROVED** [review]
